### DISCUSSION TITLE
How does Fabric. Js dynamically display the grid

### DISCUSSION THREAD

**Discussion by Ezr1al** [fabricjs/fabric.js#6914] [Q&A]
I want to display the grid in the canvas. When the canvas is panned and zoomed, the grid can also be displayed normally. Is there any good solution?I am now combining horizontal and vertical lines to form a grid, which is too big to draw, so I am only drawing the grid within the screen.How to display the mesh as the screen moves and zooms in and out.

By default:
![image](https://user-images.githubusercontent.com/36941749/110104803-93c25700-7de2-11eb-96a1-d920977f54a5.png)

When you move the canvas around:
![image](https://user-images.githubusercontent.com/36941749/110104879-ab99db00-7de2-11eb-914a-ebb081bd6365.png)

**Comment by asturur** [comment]
How do you want this grid to behave?

**Comment by mitkaaa** [comment]
This is my decision https://codepen.io/mitka3000/pen/poKadKr

But now I'm looking for a solution for a blurry background at high zoom

**Reply by rolitter** [reply]
you can set other layerfor background grid
like this
https://codepen.io/rolitter/pen/gOKxExx

**Reply by mitkaaa** [reply]
I did it https://codepen.io/mitka3000/pen/poKadKr

The grid adjusts to your zoom

**Comment by hihilisk** [comment]
try add css
```CSS
    background: repeating-linear-gradient(transparent, transparent 10px, rgba(0, 0, 0, 0.1) 11px), repeating-linear-gradient(90deg, transparent, transparent 10px, rgba(0, 0, 0, 0.1) 11px);
```
for `.lower-canvas`

**Reply by mitkaaa** [reply]
This is not my way
I am planning to use fabric on React Native.
